### DISCUSSION TITLE
is it worth it to fix the order of stats displayed in the pbar?

### DISCUSSION THREAD

**Discussion by radekosmulski** [pytorch/ignite#1936] [Ideas]
I would probably prefer if the stats displayed on the pbar followed the order I specify. This way I could keep the same order across projects, I could standardize on having the running average of the loss over be displayed first, etc.

It would probably make it easier for me to scan to see how training is going.

Currently, metrics are not displayed in the order they are added (and there is no other way to control their order):
![image](https://user-images.githubusercontent.com/2444926/114546532-8b6cff80-9c5d-11eb-8c49-38e1e42abf73.png)

To change this, all we would need to do is to change this line:
```
        if rendered_metrics:
            logger.pbar.set_postfix(**rendered_metrics)  # type: ignore[attr-defined]
```
to 
```
        if rendered_metrics:
            logger.pbar.set_postfix(rendered_metrics)  # type: ignore[attr-defined]
```
and this is what we would get:
![image](https://user-images.githubusercontent.com/2444926/114547606-ec490780-9c5e-11eb-92f2-0d548ef42b92.png)

Could I please ask if this is something worth making a change to? If so, I can open a PR :slightly_smiling_face: 

Question - would it make sense to discuss such ideas here or would it be better if I were to open an issue?

**Comment by vfdev-5** [comment]
@radekosmulski thanks for the suggestion. Yes, I think this can make perfect sense and can be integrated. Please send a PR with the fix 👍  

> Question - would it make sense to discuss such ideas here or would it be better if I were to open an issue?

You can ask here or open an issue, both works. Maybe, issues are more old fashioned and people look more there than here.

**Reply by radekosmulski** [reply]
Thank you very much for your reply @vfdev-5! will open a PR :slightly_smiling_face:
